packages/worker-relay/src/sqlite/sqlite-relay.ts

```diff
@@ -132,69 +132,82 @@ export class SqliteRelay extends EventEmitter<RelayHandlerEvents> implements Rel
   #insertEvent(db: Database, ev: NostrEvent) {
     if (this.#seenInserts.has(ev.id)) return false;
 
-    const legacyReplacable = [0, 3, 41];
-    if (legacyReplacable.includes(ev.kind) || (ev.kind >= 10_000 && ev.kind < 20_000)) {
-      const oldEvents = db.selectValues("select id from events where kind = ? and pubkey = ? and created <= ?", [
-        ev.kind,
-        ev.pubkey,
-        ev.created_at,
-      ]) as Array<string>;
+    const legacyReplaceableKinds = [0, 3, 41];
+
+    // Handle legacy and standard replaceable events (kinds 0, 3, 41, 10000-19999)
+    if (legacyReplaceableKinds.includes(ev.kind) || (ev.kind >= 10_000 && ev.kind < 20_000)) {
+      const oldEvents = db.selectValues(
+        `SELECT id FROM events WHERE kind = ? AND pubkey = ?`,
+        [ev.kind, ev.pubkey]
+      ) as Array<string>;
+
       if (oldEvents.includes(ev.id)) {
-        // we already have this event, return
+        // Already have this event
         this.#seenInserts.add(ev.id);
-        if (oldEvents.length > 1) {
```

Comment:
@v0l This was causing problems for me, but you likely had it here for a reason...

Reply:
There was a bug, there was multiple copies of the RE's and they didn't get deleted because there was no newer version to replace them.

The `> 1` check was there to delete the older versions, maybe its not needed now because most should have this old data cleaned up by now